pkg/app/server/grpcapi/web_api.go

```diff
@@ -1329,32 +1334,94 @@ func (a *WebAPI) GetMe(ctx context.Context, req *webservice.GetMeRequest) (*webs
 	}, nil
 }
 
-func (a *WebAPI) ListProjectRBACRoles(_ context.Context, _ *webservice.ListProjectRBACRolesRequest) (*webservice.ListProjectRBACRolesResponse, error) {
-	return nil, status.Errorf(codes.Unimplemented, "not implemented")
-}
+func (a *WebAPI) AddProjectRBACRole(ctx context.Context, req *webservice.AddProjectRBACRoleRequest) (*webservice.AddProjectRBACRoleResponse, error) {
+	claims, err := rpcauth.ExtractClaims(ctx)
+	if err != nil {
+		a.logger.Error("failed to authenticate the current user", zap.Error(err))
+		return nil, err
+	}
 
-func (a *WebAPI) AddProjectRBACRole(_ context.Context, _ *webservice.AddProjectRBACRoleRequest) (*webservice.AddProjectRBACRoleResponse, error) {
-	return nil, status.Errorf(codes.Unimplemented, "not implemented")
-}
+	if _, ok := a.projectsInConfig[claims.Role.ProjectId]; ok {
+		return nil, status.Error(codes.FailedPrecondition, "Failed to update a debug project specified in the control-plane configuration")
+	}
 
-func (a *WebAPI) UpdateProjectRBACRole(_ context.Context, _ *webservice.UpdateProjectRBACRoleRequest) (*webservice.UpdateProjectRBACRoleResponse, error) {
-	return nil, status.Errorf(codes.Unimplemented, "not implemented")
+	if err := a.projectStore.AddProjectRBACRole(ctx, claims.Role.ProjectId, req.Name, req.Policies); err != nil {
+		a.logger.Error("failed to add rbac role", zap.Error(err))
+		return nil, status.Error(codes.Internal, "Failed to add rbac role")
+	}
+	return &webservice.AddProjectRBACRoleResponse{}, nil
 }
 
-func (a *WebAPI) DeleteProjectRBACRole(_ context.Context, _ *webservice.DeleteProjectRBACRoleRequest) (*webservice.DeleteProjectRBACRoleResponse, error) {
-	return nil, status.Errorf(codes.Unimplemented, "not implemented")
+func (a *WebAPI) UpdateProjectRBACRole(ctx context.Context, req *webservice.UpdateProjectRBACRoleRequest) (*webservice.UpdateProjectRBACRoleResponse, error) {
+	claims, err := rpcauth.ExtractClaims(ctx)
+	if err != nil {
+		a.logger.Error("failed to authenticate the current user", zap.Error(err))
+		return nil, err
+	}
+
+	if _, ok := a.projectsInConfig[claims.Role.ProjectId]; ok {
+		return nil, status.Error(codes.FailedPrecondition, "Failed to update a debug project specified in the control-plane configuration")
+	}
+
+	if err := a.projectStore.UpdateProjectRBACRole(ctx, claims.Role.ProjectId, req.Name, req.Policies); err != nil {
+		a.logger.Error("failed to update rbac role", zap.Error(err))
+		return nil, status.Error(codes.Internal, "Failed to update rbac role")
```

Comment:
Should we use `gRPCEntityOperationError` to have a correct error code?

Reply:
Other places as well.

Reply:
True,  let me fix this.